kobo/django/views/generic.py

```diff
@@ -3,10 +3,54 @@
 import warnings
 from django.conf import settings
 from django.db.models.query import QuerySet
+from django.http import HttpResponse
+from django.template.loader import get_template
 from django.views.generic.edit import ProcessFormView, FormMixin
 from django.views.generic.list import ListView
 from django.views.generic.detail import DetailView
 
+
+class UsersAclMixin:
+    def _check_acl_permission(self, request):
```

Comment:
I wonder if you could summarize what this method is supposed to return in a comment or doc string?

Right now the design is a little confusing to me.  This method seems to be:

Returns a string summarizing the reason why access is denied to the user list, e.g. "staff" if access is denied because user is not staff; or returns blank if access *is* permitted.

But it's kind of unclear, because elsewhere you take the return value of this method and then do some additional checks afterward, which would seem to imply that the responsibility for deciding whether access is permitted is not solely the responsibility of this method.  If that were true then I don't understand what's the meaning of this method.

Put another way: can you explain clearly what the return value is supposed to represent?

Reply:
correct, it is indeed confusing, thanks for the suggestion. I'll amend and make it more concise. i.e., rename it to from `_check_acl_permission` to `_has_access` which returns True/False for brevity.